src/NonVariantWrappers.dfy

```diff
@@ -0,0 +1,109 @@
+// RUN: %dafny /compile:0 "%s"
+
+// Assert that this file is minimally different from Wrappers.dfy,
+// to ensure they are mostly drop-in replacements for each other.
+// RUN: %diff "%S/Wrappers.dfy" "%s" | tail -n +3 > "%t" || true
+// RUN: %diff "%t" "%s".diff.expect
+
+/*******************************************************************************
+*  Copyright by the contributors to the Dafny Project
+*  SPDX-License-Identifier: MIT 
+*******************************************************************************/
+
+// Copy of Wrappers.dfy without covariance on type parameters.
+// This version can be useful to work around the lack of variance support
+// for particular compilation targets, especially since variance
+// is not necessary for these datatypes most of the time.
```

Comment:
"most" of the time may be a bit strong (since variance is needed to work with subset types, for example)

Reply:
Dialled the statement down a little, but also added an example file to illustrate why covariance often isn't necessary.